tests/vertexonly/test_vertex_only_fs.py

```diff
@@ -68,6 +68,9 @@ def functionspace_tests(vm):
     # Can interpolate and Galerkin project expressions onto functions
     f.interpolate(expr)
     g.project(expr)
+    # Empty halos for functions on vertex only mesh
+    assert np.allclose(f.dat.data_ro, f.dat.data_ro_with_halos)
+    assert np.allclose(g.dat.data_ro, g.dat.data_ro_with_halos)
```

Comment:
Can you assert that the sets have the right shape. `f.dof_dset.sizes` should be 3 numbers, all the same.

Reply:
Added! Thanks